cre.cpp

```diff
@@ -1392,6 +1392,48 @@ static int isXPointerInCurrentPage(lua_State *L) {
 	return 1;
 }
 
+static int hasXPointer(lua_State *L) {
+    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
+    const char *xpointer_str = luaL_checkstring(L, 2);
+
+    bool found = true;
+    ldomXPointer xp = doc->dom_doc->createXPointer(lString16(xpointer_str));
+    lvPoint pt = xp.toPoint();
+    if (pt.y < 0) {
+        found = false;
+    }
+    lua_pushboolean(L, found);
+    return 1;
+}
+
+static int getImageDataFromPosition(lua_State *L) {
+    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
+    int x = luaL_checkint(L, 2);
+    int y = luaL_checkint(L, 3);
+    lvPoint pt(x, y);
+    ldomXPointer ptr = doc->text_view->getNodeByPoint(pt);
+    if (ptr.isNull())
+        return 0;
+    LVStreamRef stream = ptr.getNode()->getObjectImageStream();
+    if (!stream.isNull()) {
+        unsigned size = stream->GetSize();
+        lvsize_t read_size = 0;
+        void *buffer = (void *)malloc(size);
```

Comment:
Well, my first attempt code was simpler:
```
 /* This version would not need that and buffer would be freed by lua GC */
 void *buffer = lua_newuserdata(L, size);
 if (buffer != NULL) {
     stream->Read(buffer, size, &read_size);
     if (read_size == size) {
         lua_pushinteger(L, size);
         return 2;
     }
 }
```
Without any metatable stuff. I tested that (1000 calls from lua, no memory leak). Does that look fine to you ?
But as i'm not really at ease with C, I thought i'll do just like it is done in getCoverPageImageData() (with lightuserdata and ffi.C.free in lua code), to be conservative and avoid discussion on a topic i'm not confortable with :)
You tell me.

Btw, when reading the old issues about BlitBuffer leaks, I got the impression LUAJIT GC couldn't be trusted with them, and that is why we have to add explicite _bb:free() like i'm doing in ImageWidget.
(Or is there some C++ vs C differences here ?)